sanic/config.py

```diff
@@ -50,9 +57,17 @@ def __init__(self, defaults=None, load_env=True, keep_alive=None):
         if keep_alive is not None:
             self.KEEP_ALIVE = keep_alive
 
-        if load_env:
-            prefix = SANIC_PREFIX if load_env is True else load_env
-            self.load_environment_vars(prefix=prefix)
+        if load_env is not True:
+            if load_env:
+                self.load_environment_vars(prefix=load_env)
+            warn(
+                "Use of load_env is deprecated and will be removed in "
+                "21.12. Modify the configuration prefix by passing "
+                "env_prefix instead.",
+                DeprecationWarning,
+            )
+        else:
```

Comment:
I think this should be:

```python
elif env_prefix:
```

Reply:
And then add one more test with:

```python
Sanic(..., env_prefix=None)
```

where the variable is not loaded. You probably need to add a catch for this in the above line if they are still using `load_env` though.

OR, if `env_prefix != SANIC_PREFIX` then skip the whole `load_env` part altogether.

Reply:
:heavy_check_mark: